doc/design/wallet/wallet-design.md

```diff
@@ -0,0 +1,768 @@
+                                      options_json: *const c_char,
+                                      cb: Option<extern fn(command_handle_: i32, err: ErrorCode,
+                                                           wallet_search_handle_p: *mut i32)>) -> ErrorCode {}
+
+
+/// Fetch next records for wallet search.
+///
+/// Not if there are no records this call returns WalletNoRecords error.
+///
+/// #Params
+/// wallet_handle: wallet handle (created by open_wallet)
+/// wallet_search_handle: wallet wallet handle (created by indy_open_wallet_search)
+/// count: Count of records to fetch
+///
+/// #Returns
+/// wallet records json:
+/// {
+///   totalCount: <int>, // present only if retrieveTotalCount set to true
+///   records: [{ // present only if retrieveRecords set to true
+///       id: "Some id",
+///       value: "Some value", // present only if retrieveValue set to true
+///       tags: <tags json>, // present only if retrieveTags set to true
+///   }],
+/// }
+extern pub fn indy_fetch_wallet_search_next_records(command_handle: u32,
+                                                    wallet_handle: i32,
+                                                    wallet_search_handle: i32,
+                                                    count: u32,
+                                                    cb: Option<extern fn(command_handle_: i32, err: ErrorCode,
+                                                                         records_json: *const c_char)>) -> ErrorCode {}
+```
+
+## Wallet API and Storage Interface
+
+Wallet API already exists and allows wallet management. It requires to be update to allow plugging different storages implementation to libindy. For example, to allow storing
+of wallet records in SQL database. To achieve this we will replace existing ```indy_register_wallet_type``` call with ```indy_register_wallet_storage``` call:
+
+```Rust
+/// Register custom wallet storage implementation.
+///
+/// It allows library user to provide custom wallet storage implementation as set of handlers.
+///
+/// #Params
+/// command_handle: Command handle to map callback to caller context.
+/// type_: Wallet storage name.
+/// create: Wallet storage "create" operation handler
+/// delete: Wallet storage "delete" operation handler
+/// open: Wallet storage "open" operation handler
+/// close: Wallet storage "close" operation handler
+/// add_record: Wallet storage "add_record" operation handler
+/// update_record_value: Wallet storage "update_record_value" operation handler
+/// update_record_tags: Wallet storage "update_record_tags" operation handler
+/// add_record_tags: Wallet storage "add_record_tags" operation handler
+/// delete_record_tags: Wallet storage "delete_record_tags" operation handler
+/// delete_record: Wallet storage "delete_record" operation handler
+/// get_record: Wallet storage "get_record" operation handler
+/// get_record_id: Wallet storage "get_record_id" operation handler
```

Comment:
get_record_type would also be needed for export.

Reply:
Got it, will provide solution